graphql-dgs-codegen-gradle/src/test/kotlin/com/netflix/graphql/dgs/CodegenGradlePluginTest.kt

```diff
@@ -23,19 +23,32 @@ package com.netflix.graphql.dgs
 import org.assertj.core.api.Assertions.assertThat
 import org.gradle.testkit.runner.GradleRunner
 import org.gradle.testkit.runner.TaskOutcome.SUCCESS
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
-import java.io.File
+import org.junit.jupiter.api.io.TempDir
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.div
 
 /**
  * A simple unit test for the 'com.netflix.graphql.dgs.greeting' plugin.
  */
 class CodegenGradlePluginTest {
 
+    @TempDir
+    lateinit var projectDir: Path
+
+    @BeforeEach
+    fun setUp() {
+        val projectResourcePath = Paths.get("src", "test", "resources", "test-project")
+        projectResourcePath.toFile().copyRecursively(target = projectDir.toFile())
```

Comment:
The tests were writing to the resources directory, which could make them nondeterministic. I switched them to use a temporary directory and copy the project files into it between each test case.